geo_map/static/geo_map/index.js

```diff
@@ -190,11 +193,14 @@ function fetchDataAndCreateMap(
       const centerCoordinates = calculateCenter(data);
       const mapOptions = {
         center: centerCoordinates,
-        zoom: 6,
+        zoom: currentZoom,
         mapId: "Netbox_MAP_ID",
       };
       map = new google.maps.Map(document.getElementById("map"), mapOptions);
-      zoom = map.getZoom();
+      if (currentCenter) {
+        map.setCenter(currentCenter);
+      }
```

Comment:
you can just set it in the `mapOptions` on line 195

Reply:
With this change, the center isn't correctly saved when preselecting.